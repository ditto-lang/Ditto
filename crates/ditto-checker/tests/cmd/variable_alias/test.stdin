module Test exports (..);

type alias WhyTho(a) = a;

identity = fn (a: WhyTho(a)) -> a;

five : Float = identity(5.0);
