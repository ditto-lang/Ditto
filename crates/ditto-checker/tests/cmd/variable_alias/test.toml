bin.name = "ditto-checker-testbin-check-module"
args = ["variable_alias.ditto"]
fs.sandbox = true
