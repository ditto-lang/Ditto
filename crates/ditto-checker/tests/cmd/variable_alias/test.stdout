{
  "module_name": [
    "Test"
  ],
  "exports": {
    "types": {
      "WhyTho": {
        "Alias": {
          "doc_comments": [],
          "doc_position": 0,
          "kind": {
            "Function": {
              "parameters": [
                "Type"
              ]
            }
          },
          "aliased_type": {
            "type": "Variable",
            "data": {
              "variable_kind": "Type",
              "var": 0,
              "source_name": "a"
            }
          },
          "alias_variables": [
            0
          ]
        }
      }
    },
    "constructors": {},
    "values": {
      "five": {
        "doc_comments": [],
        "doc_position": 0,
        "value_type": {
          "type": "Call",
          "data": {
            "function": {
              "type": "ConstructorAlias",
              "data": {
                "constructor_kind": {
                  "Function": {
                    "parameters": [
                      "Type"
                    ]
                  }
                },
                "canonical_value": {
                  "module_name": [
                    null,
                    [
                      "Test"
                    ]
                  ],
                  "value": "WhyTho"
                },
                "source_value": {
                  "module_name": null,
                  "value": "WhyTho"
                },
                "alias_variables": [
                  0
                ],
                "aliased_type": {
                  "type": "PrimConstructor",
                  "data": "Float"
                }
              }
            },
            "arguments": [
              {
                "type": "PrimConstructor",
                "data": "Float"
              }
            ]
          }
        }
      },
      "identity": {
        "doc_comments": [],
        "doc_position": 1,
        "value_type": {
          "type": "Function",
          "data": {
            "parameters": [
              {
                "type": "Call",
                "data": {
                  "function": {
                    "type": "ConstructorAlias",
                    "data": {
                      "constructor_kind": {
                        "Function": {
                          "parameters": [
                            "Type"
                          ]
                        }
                      },
                      "canonical_value": {
                        "module_name": [
                          null,
                          [
                            "Test"
                          ]
                        ],
                        "value": "WhyTho"
                      },
                      "source_value": {
                        "module_name": null,
                        "value": "WhyTho"
                      },
                      "alias_variables": [
                        0
                      ],
                      "aliased_type": {
                        "type": "Variable",
                        "data": {
                          "variable_kind": "Type",
                          "var": 0,
                          "source_name": "a"
                        }
                      }
                    }
                  },
                  "arguments": [
                    {
                      "type": "Variable",
                      "data": {
                        "variable_kind": "Type",
                        "var": 0,
                        "source_name": "a"
                      }
                    }
                  ]
                }
              }
            ],
            "return_type": {
              "type": "Call",
              "data": {
                "function": {
                  "type": "ConstructorAlias",
                  "data": {
                    "constructor_kind": {
                      "Function": {
                        "parameters": [
                          "Type"
                        ]
                      }
                    },
                    "canonical_value": {
                      "module_name": [
                        null,
                        [
                          "Test"
                        ]
                      ],
                      "value": "WhyTho"
                    },
                    "source_value": {
                      "module_name": null,
                      "value": "WhyTho"
                    },
                    "alias_variables": [
                      0
                    ],
                    "aliased_type": {
                      "type": "Variable",
                      "data": {
                        "variable_kind": "Type",
                        "var": 0,
                        "source_name": "a"
                      }
                    }
                  }
                },
                "arguments": [
                  {
                    "type": "Variable",
                    "data": {
                      "variable_kind": "Type",
                      "var": 0,
                      "source_name": "a"
                    }
                  }
                ]
              }
            }
          }
        }
      }
    }
  },
  "types": {
    "WhyTho": {
      "Alias": {
        "doc_comments": [],
        "type_name_span": {
          "start_offset": 38,
          "end_offset": 44
        },
        "kind": {
          "Function": {
            "parameters": [
              "Type"
            ]
          }
        },
        "aliased_type": {
          "type": "Variable",
          "data": {
            "variable_kind": "Type",
            "var": 0,
            "source_name": "a"
          }
        },
        "alias_variables": [
          0
        ]
      }
    }
  },
  "constructors": {},
  "values": {
    "identity": {
      "doc_comments": [],
      "name_span": {
        "start_offset": 54,
        "end_offset": 62
      },
      "expression": {
        "expression": "Function",
        "data": {
          "span": {
            "start_offset": 65,
            "end_offset": 87
          },
          "function_type": {
            "type": "Function",
            "data": {
              "parameters": [
                {
                  "type": "Call",
                  "data": {
                    "function": {
                      "type": "ConstructorAlias",
                      "data": {
                        "constructor_kind": {
                          "Function": {
                            "parameters": [
                              "Type"
                            ]
                          }
                        },
                        "canonical_value": {
                          "module_name": [
                            null,
                            [
                              "Test"
                            ]
                          ],
                          "value": "WhyTho"
                        },
                        "source_value": {
                          "module_name": null,
                          "value": "WhyTho"
                        },
                        "alias_variables": [
                          0
                        ],
                        "aliased_type": {
                          "type": "Variable",
                          "data": {
                            "variable_kind": "Type",
                            "var": 0,
                            "source_name": "a"
                          }
                        }
                      }
                    },
                    "arguments": [
                      {
                        "type": "Variable",
                        "data": {
                          "variable_kind": "Type",
                          "var": 0,
                          "source_name": "a"
                        }
                      }
                    ]
                  }
                }
              ],
              "return_type": {
                "type": "Call",
                "data": {
                  "function": {
                    "type": "ConstructorAlias",
                    "data": {
                      "constructor_kind": {
                        "Function": {
                          "parameters": [
                            "Type"
                          ]
                        }
                      },
                      "canonical_value": {
                        "module_name": [
                          null,
                          [
                            "Test"
                          ]
                        ],
                        "value": "WhyTho"
                      },
                      "source_value": {
                        "module_name": null,
                        "value": "WhyTho"
                      },
                      "alias_variables": [
                        0
                      ],
                      "aliased_type": {
                        "type": "Variable",
                        "data": {
                          "variable_kind": "Type",
                          "var": 0,
                          "source_name": "a"
                        }
                      }
                    }
                  },
                  "arguments": [
                    {
                      "type": "Variable",
                      "data": {
                        "variable_kind": "Type",
                        "var": 0,
                        "source_name": "a"
                      }
                    }
                  ]
                }
              }
            }
          },
          "binders": [
            [
              {
                "Variable": {
                  "span": {
                    "start_offset": 69,
                    "end_offset": 70
                  },
                  "name": "a"
                }
              },
              {
                "type": "Call",
                "data": {
                  "function": {
                    "type": "ConstructorAlias",
                    "data": {
                      "constructor_kind": {
                        "Function": {
                          "parameters": [
                            "Type"
                          ]
                        }
                      },
                      "canonical_value": {
                        "module_name": [
                          null,
                          [
                            "Test"
                          ]
                        ],
                        "value": "WhyTho"
                      },
                      "source_value": {
                        "module_name": null,
                        "value": "WhyTho"
                      },
                      "alias_variables": [
                        0
                      ],
                      "aliased_type": {
                        "type": "Variable",
                        "data": {
                          "variable_kind": "Type",
                          "var": 0,
                          "source_name": "a"
                        }
                      }
                    }
                  },
                  "arguments": [
                    {
                      "type": "Variable",
                      "data": {
                        "variable_kind": "Type",
                        "var": 0,
                        "source_name": "a"
                      }
                    }
                  ]
                }
              }
            ]
          ],
          "body": {
            "expression": "LocalVariable",
            "data": {
              "span": {
                "start_offset": 86,
                "end_offset": 87
              },
              "variable_type": {
                "type": "Call",
                "data": {
                  "function": {
                    "type": "ConstructorAlias",
                    "data": {
                      "constructor_kind": {
                        "Function": {
                          "parameters": [
                            "Type"
                          ]
                        }
                      },
                      "canonical_value": {
                        "module_name": [
                          null,
                          [
                            "Test"
                          ]
                        ],
                        "value": "WhyTho"
                      },
                      "source_value": {
                        "module_name": null,
                        "value": "WhyTho"
                      },
                      "alias_variables": [
                        0
                      ],
                      "aliased_type": {
                        "type": "Variable",
                        "data": {
                          "variable_kind": "Type",
                          "var": 0,
                          "source_name": "a"
                        }
                      }
                    }
                  },
                  "arguments": [
                    {
                      "type": "Variable",
                      "data": {
                        "variable_kind": "Type",
                        "var": 0,
                        "source_name": "a"
                      }
                    }
                  ]
                }
              },
              "variable": "a"
            }
          }
        }
      }
    },
    "five": {
      "doc_comments": [],
      "name_span": {
        "start_offset": 90,
        "end_offset": 94
      },
      "expression": {
        "expression": "Call",
        "data": {
          "span": {
            "start_offset": 105,
            "end_offset": 118
          },
          "call_type": {
            "type": "Call",
            "data": {
              "function": {
                "type": "ConstructorAlias",
                "data": {
                  "constructor_kind": {
                    "Function": {
                      "parameters": [
                        "Type"
                      ]
                    }
                  },
                  "canonical_value": {
                    "module_name": [
                      null,
                      [
                        "Test"
                      ]
                    ],
                    "value": "WhyTho"
                  },
                  "source_value": {
                    "module_name": null,
                    "value": "WhyTho"
                  },
                  "alias_variables": [
                    0
                  ],
                  "aliased_type": {
                    "type": "PrimConstructor",
                    "data": "Float"
                  }
                }
              },
              "arguments": [
                {
                  "type": "PrimConstructor",
                  "data": "Float"
                }
              ]
            }
          },
          "function": {
            "expression": "LocalVariable",
            "data": {
              "span": {
                "start_offset": 105,
                "end_offset": 113
              },
              "variable_type": {
                "type": "Function",
                "data": {
                  "parameters": [
                    {
                      "type": "Call",
                      "data": {
                        "function": {
                          "type": "ConstructorAlias",
                          "data": {
                            "constructor_kind": {
                              "Function": {
                                "parameters": [
                                  "Type"
                                ]
                              }
                            },
                            "canonical_value": {
                              "module_name": [
                                null,
                                [
                                  "Test"
                                ]
                              ],
                              "value": "WhyTho"
                            },
                            "source_value": {
                              "module_name": null,
                              "value": "WhyTho"
                            },
                            "alias_variables": [
                              0
                            ],
                            "aliased_type": {
                              "type": "PrimConstructor",
                              "data": "Float"
                            }
                          }
                        },
                        "arguments": [
                          {
                            "type": "PrimConstructor",
                            "data": "Float"
                          }
                        ]
                      }
                    }
                  ],
                  "return_type": {
                    "type": "Call",
                    "data": {
                      "function": {
                        "type": "ConstructorAlias",
                        "data": {
                          "constructor_kind": {
                            "Function": {
                              "parameters": [
                                "Type"
                              ]
                            }
                          },
                          "canonical_value": {
                            "module_name": [
                              null,
                              [
                                "Test"
                              ]
                            ],
                            "value": "WhyTho"
                          },
                          "source_value": {
                            "module_name": null,
                            "value": "WhyTho"
                          },
                          "alias_variables": [
                            0
                          ],
                          "aliased_type": {
                            "type": "PrimConstructor",
                            "data": "Float"
                          }
                        }
                      },
                      "arguments": [
                        {
                          "type": "PrimConstructor",
                          "data": "Float"
                        }
                      ]
                    }
                  }
                }
              },
              "variable": "identity"
            }
          },
          "arguments": [
            {
              "Expression": {
                "expression": "Float",
                "data": {
                  "span": {
                    "start_offset": 114,
                    "end_offset": 117
                  },
                  "value": "5.0",
                  "value_type": {
                    "type": "Call",
                    "data": {
                      "function": {
                        "type": "ConstructorAlias",
                        "data": {
                          "constructor_kind": {
                            "Function": {
                              "parameters": [
                                "Type"
                              ]
                            }
                          },
                          "canonical_value": {
                            "module_name": [
                              null,
                              [
                                "Test"
                              ]
                            ],
                            "value": "WhyTho"
                          },
                          "source_value": {
                            "module_name": null,
                            "value": "WhyTho"
                          },
                          "alias_variables": [
                            0
                          ],
                          "aliased_type": {
                            "type": "PrimConstructor",
                            "data": "Float"
                          }
                        }
                      },
                      "arguments": [
                        {
                          "type": "PrimConstructor",
                          "data": "Float"
                        }
                      ]
                    }
                  }
                }
              }
            }
          ]
        }
      }
    }
  },
  "values_toposort": [
    "identity",
    "five"
  ]
}